### PR TITLE
SO-60

### PR DESCRIPTION
**Pull request by Miguel1357**
- Added send password reset email link
- Passwords can now be updated
- Component for recover password and update password
- Tested cases for password update errors

### CODE REVIEW

**Comment by Miguel1357** [comment]
TO DO: 
1. Test and create more popups for more password update error cases in UpdatePasswordForm.tsx (no password repeats, minimum password requirements validation, too many requests).
2. The code can be a bit janky at times. It is still possible to receive "Auth Session Missing".
3. If the user receives an error at the password update page and they reload the page, then they will be automatically taken to the dashboard. This can be a bit confusing for the user who will believe that they were never logged in despite secretly logging in when clicking the URL sent to them through their email.
4. Sometimes the send reset link page may say that a link has been successfully sent even though the email was never received by the user. This may be more of a Supabase issue rather than an issue with our own product, but it should still be looked into.

File: src/app/(auth)/base.css [hunk 1]
```diff
@@ -66,4 +66,11 @@
 		/* Styling (Background) */
 		@apply bg-[#5A8472];
 	}
+	#reset-link-span,
```
Comment (jvniorrr) [inline src/app/(auth)/base.css]: Unsure if this code is being used anywhere?

If not remove this block scope.

File: src/app/(auth)/recoverpassword/page.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+// meta tags
+import type { Metadata } from 'next';
+export const metadata: Metadata = {
+	title: 'Recover Password Page',
+	description: 'Join us at CAMEL.',
```
Comment (jvniorrr) [inline src/app/(auth)/recoverpassword/page.tsx]: Change description for this metadata description. Not necessarily required just should align with page if you include such pages..

Same idea follows for both the /recoverpassword route and the /updatepassword route pages. 

"**CAMEL - Password Reset Page**" or "**CAMEL - Securely Reset Your Password**" or something

File: src/components/UpdatePassword/UpdatePasswordForm.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+			if (error) {
+				setPopupMessage(
+					'Error updating password. Please try again later.',
+				);
+				setTimeout(() => {
+					setSubmitted(false);
+					router.push('/recoverpassword');
+				}, 3000);
+			}
+			console.log(data);
```
Comment (jvniorrr) [inline src/components/UpdatePassword/UpdatePasswordForm.tsx]: remove this console.log and on line 73 as well. :)

**Review by jvniorrr — COMMENTED** [review]
Minor changes needed. 👍 
Good code, very nice work!!! 

Good Work!

File: src/app/(auth)/base.css (hunk 1 above)
Comment (hashemJaber) [reply src/app/(auth)/base.css]: Was this resolved Miguel ?

Comment (Miguel1357) [reply src/app/(auth)/base.css]: This code is not needed. I am removing it.

File: src/app/(auth)/recoverpassword/page.tsx (hunk 2 above)
Comment (Miguel1357) [reply src/app/(auth)/recoverpassword/page.tsx]: Descriptions are being taken care of to fit pages.

File: src/components/UpdatePassword/UpdatePasswordForm.tsx (hunk 3 above)
Comment (Miguel1357) [reply src/components/UpdatePassword/UpdatePasswordForm.tsx]: Console.logs for lines 73 and 83 being removed.

**Review by jvniorrr — APPROVED** [review]

**Comment by jvniorrr** [comment]
Ready to rebase and merge in my eyes. Thank you kind sir! Awesome work once again! >:)
Merge Accepted

**Pull request merged**